### PR TITLE
TextArea, SelectList: Add on prop and update lightBackground/darkBackground colors

### PR DESCRIPTION
**Pull request by jliotta**
[SYN-79](https://cambly.atlassian.net/browse/SYN-79)

This PR accomplishes the following:
- updates text, border colors for lightBackground/darkBackground and selected states for `TextField`
- adds `on` prop to `TextArea` and `SelectList`
- updates stories for all three components

TextField lightBackground:

https://github.com/user-attachments/assets/e0b84d29-b767-49a9-a6ef-54b7ee6fbee8

TextField lightBackground error state:

https://github.com/user-attachments/assets/b32ca24b-fff6-4131-9a7b-1775fd9351da

 TextField darkBackground and error state:

https://github.com/user-attachments/assets/0d7ac821-402f-41c9-890d-6e1043a4deb8

TextArea lightBackground and error state:

https://github.com/user-attachments/assets/1100ab15-81f6-4a13-af98-0255e797a7c8

TextArea darkBackground and error state:

https://github.com/user-attachments/assets/cf2d7760-ae52-48a5-ad5f-9066427a21db

SelectList lightBackground and error state:

https://github.com/user-attachments/assets/19990103-f763-4d0f-aa03-6fdf31199154

SelectList darkBackground and error state:

https://github.com/user-attachments/assets/c53b7b57-d27d-4c1b-85dd-5a9855771047

[SYN-79]: https://cambly.atlassian.net/browse/SYN-79?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/small-apes-explode.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@cambly/syntax-core": major
+---
+
+TextArea, SelectList: Add on prop and update lightBackground/darkBackgroudn colors
```
Comment (zprong) [inline .changeset/small-apes-explode.md:5]: typo nit:
```suggestion
TextArea, SelectList: Add on prop and update lightBackground/darkBackground colors
```

File: packages/syntax-core/src/TextArea/TextArea.tsx [hunk 2]
```diff
@@ -88,6 +95,11 @@ const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
     const disabled = !isHydrated || disabledProp;
     const reactId = useId();
     const inputId = id ?? reactId;
+    const textColor = on === "darkBackground" ? "white" : "gray900";
```
Comment (zprong) [inline packages/syntax-core/src/TextArea/TextArea.tsx:98]: the original inlined color  was `gray700`, is this intentional?

**Review by zprong — APPROVED** [review]
LGTM!

**Review by btorndorff — APPROVED** [review]
LG

File: packages/syntax-core/src/TextArea/TextArea.tsx (hunk 2 above)
Comment (jliotta) [reply packages/syntax-core/src/TextArea/TextArea.tsx:98]: Yep! These should all be gray900 and match what we have in `TextField`

**Pull request merged**